ra/ra.go

```diff
@@ -494,59 +514,106 @@ func (ra *RegistrationAuthorityImpl) NewAuthorization(ctx context.Context, reque
 	}
 
 	if ra.reuseValidAuthz {
-		auths, err := ra.SA.GetValidAuthorizations(ctx, regID, []string{identifier.Value}, ra.clk.Now())
-		if err != nil {
-			outErr := berrors.InternalServerError(
-				"unable to get existing validations for regID: %d, identifier: %s, %s",
-				regID,
-				identifier.Value,
-				err,
-			)
-			ra.log.Warning(outErr.Error())
-			return core.Authorization{}, outErr
-		}
-
-		if existingAuthz, ok := auths[identifier.Value]; ok {
-			// Use the valid existing authorization's ID to find a fully populated version
-			// The results from `GetValidAuthorizations` are most notably missing
-			// `Challenge` values that the client expects in the result.
-			populatedAuthz, err := ra.SA.GetAuthorization(ctx, existingAuthz.ID)
+		var auths map[string]*core.Authorization
+		var err error
+		if features.Enabled(features.NewAuthorizationSchema) {
+			now := ra.clk.Now().UnixNano()
+			authzMapPB, err := ra.SA.GetValidAuthorizations2(ctx, &sapb.GetValidAuthorizationsRequest{
+				RegistrationID: &regID,
+				Domains:        []string{identifier.Value},
+				Now:            &now,
+			})
+			if err != nil {
+				outErr := berrors.InternalServerError(
+					"unable to get existing validations for regID: %d, identifier: %s, %s",
+					regID,
+					identifier.Value,
+					err,
+				)
+				ra.log.Warning(outErr.Error())
+				return core.Authorization{}, outErr
+			}
+			auths, err = bgrpc.PBToAuthzMap(authzMapPB)
+			if err != nil {
+				return core.Authorization{}, err
+			}
+		} else {
+			auths, err = ra.SA.GetValidAuthorizations(ctx, regID, []string{identifier.Value}, ra.clk.Now())
 			if err != nil {
 				outErr := berrors.InternalServerError(
-					"unable to get existing authorization for auth ID: %s",
-					existingAuthz.ID,
+					"unable to get existing validations for regID: %d, identifier: %s, %s",
+					regID,
+					identifier.Value,
+					err,
 				)
-				ra.log.Warningf("%s: %s", outErr.Error(), existingAuthz.ID)
+				ra.log.Warning(outErr.Error())
 				return core.Authorization{}, outErr
 			}
-			if ra.authzValidChallengeEnabled(&populatedAuthz) {
+		}
+
+		if existingAuthz, ok := auths[identifier.Value]; ok {
+			var authzToCheck *core.Authorization
+			if features.Enabled(features.NewAuthorizationSchema) {
+				// GetValidAuthorizations2 returns full authorizations so we
+				// don't need to do a second query to get the challenges
+				authzToCheck = existingAuthz
+			} else {
+				// Use the valid existing authorization's ID to find a fully populated version
+				// The results from `GetValidAuthorizations` are most notably missing
```

Comment:
This comment is no longer accurate. Filed https://github.com/letsencrypt/boulder/issues/4190 to clean up both the comment and the corresponding code.